grasp_selector:
  target_frame: {
    type: string,
    default_value: "world",
    read_only: true,
    description: "Name of TF frame that grasp should be expressed in.",
    validation: {
      not_empty<>: null
    }
  }
  input_topic: {
    type: string,
    default_value: "object_pose",
    read_only: true,
    description: "Name of Odometry topic tracking the object.",
    validation: {
      not_empty<>: null
    }
  }
  output_topic: {
    type: string,
    default_value: "px100/desired_eef_pose",
    read_only: true,
    description: "Name of PoseStamped topic to publish with grasp.",
    validation: {
      not_empty<>: null
    }
  }
  max_time_since_last_seen: {
    type: double,
    default_value: 1.0,
    read_only: false,
    description: "Don't try to grasp objects that are more stale than this value."
  }
  offset:
    x: {
      type: double,
      default_value: -0.05,
      read_only: false,
      description: "X-axis offset from object_pose to desired_eef_pose."
    }
    y: {
      type: double,
      default_value: -0.01,
      read_only: false,
      description: "Y-axis offset from object_pose to desired_eef_pose."
    }
    z: {
      type: double,
      default_value: 0.05,
      read_only: false,
      description: "Z-axis offset from object_pose to desired_eef_pose."
    }
